src/main/java/com/restapi/carMarket/service/CarService.java

```diff
@@ -20,68 +20,96 @@
     @Autowired
     CarDao carDao;
 
-    public void insert(Car car) {
-        if(isValid(car))
-            carDao.save(car);
-        else
-            throw new CarNotValidException();
-    }
-
-    private boolean isValid(Car car) {
-        boolean valid = true;
-        if(isValidPrice(car))
-            valid = false;
-        else if(isValidYear(car))
-            valid = false;
-        return valid;
-    }
-
-    private boolean isValidYear(Car car) {
-        return car.getYear()<1885 || car.getYear()> LocalDateTime.now().getYear();
-    }
-
-    private boolean isValidPrice(Car car) {
-        return car.getPrice()<1 || car.getPrice()>2000000000;
-    }
-
-    public List<Car> findAll() {
+    public List<Car> getAllCarsOnMarket() {
         return carDao.findAll();
     }
 
-    public Car findById(Long id) {
+    public Car getCarById(Long id) {
         Optional<Car> car = carDao.findById(id);
 
         if(car.isPresent())
             return car.get();
         else
-            throw new CarNotFoundException();
+            throw new CarNotFoundException("Car with id("+id+") is not found");
     }
 
-    public void delete(Car car) {
+    public void removeCarFromMarket(Car car) {
         if(carDao.exists(Example.of(car)))
             carDao.delete(car);
```

Comment:
I'm not familiar with `Example.of()` but I guess it queries by example. If so, it may match multiple cars, which means `carDao.delete(car)` will also delete multiple cars. Or will it delete just the first one found matching the example?

Reply:
I will try to re-rewrite the delete method on my own, so it compares objects by certain columns when tries to find it and then if found deletes it.